src/Test/Utilities/Portable/Metadata/MetadataReaderUtils.cs

```diff
@@ -430,6 +431,34 @@ internal static void VerifyMethodBodies(ImmutableArray<byte> peImage, Action<byt
 
         static readonly byte[] ThrowNull = new[] { (byte)ILOpCode.Ldnull, (byte)ILOpCode.Throw };
 
+        internal static void AssertReferencedIsReadOnlyAttributes(MetadataReader reader, CustomAttributeHandleCollection customAttributes, Accessibility accessibility)
```

Comment:
Consider making these extension methods.

Reply:
The advantage of asserting in those methods is that the test failure will give more specific information about which part of the validation failed, and while debugging it will stop at the right place too.

---
In reply to: [167377740](https://github.com/dotnet/roslyn/pull/24492#discussion_r167377740) [](ancestors = 167377740)